base/MAPL_SwathGridFactory.F90

```diff
@@ -0,0 +1,1259 @@
+      use MAPL_BaseMod
+      class(SwathGridFactory), intent(inout) :: this
+      type(ESMF_Grid),      intent(inout) :: grid
+      integer, allocatable, intent(out) :: local_start(:)
+      integer, allocatable, intent(out) :: global_start(:)
+      integer, allocatable, intent(out) :: global_count(:)
+      type(FileMetaData), intent(in), optional :: metaData
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+      integer :: global_dim(3), i1,j1,in,jn
+
+      _UNUSED_DUMMY(this)
+
+      call MAPL_GridGet(grid,globalCellCountPerDim=global_dim,rc=status)
+      _VERIFY(status)
+      call MAPL_GridGetInterior(grid,i1,in,j1,jn)
+      allocate(local_start,source=[i1,j1])
+      allocate(global_start,source=[1,1])
+      allocate(global_count,source=[global_dim(1),global_dim(2)])
+
+      _RETURN(_SUCCESS)
+
+   end subroutine generate_file_bounds
+
+   subroutine generate_file_corner_bounds(this,grid,local_start,global_start,global_count,rc)
+      use esmf
+      class (SwathGridFactory), intent(inout) :: this
+      type(ESMF_Grid), intent(inout)      :: grid
+      integer, allocatable, intent(out) :: local_start(:)
+      integer, allocatable, intent(out) :: global_start(:)
+      integer, allocatable, intent(out) :: global_count(:)
+      integer, optional, intent(out) :: rc
+
+      _UNUSED_DUMMY(this)
+      _UNUSED_DUMMY(grid)
+      _UNUSED_DUMMY(local_start)
+      _UNUSED_DUMMY(global_start)
+      _UNUSED_DUMMY(global_count)
+
+      _FAIL('unimplemented')
+      _RETURN(_SUCCESS)
+   end subroutine generate_file_corner_bounds
+
+   function generate_file_reference2D(this,fpointer) result(ref)
+      use pFIO
+      type(ArrayReference) :: ref
+      class(SwathGridFactory), intent(inout) :: this
+      real, pointer, intent(in) :: fpointer(:,:)
+      _UNUSED_DUMMY(this)
+      ref = ArrayReference(fpointer)
+   end function generate_file_reference2D
+
+   function generate_file_reference3D(this,fpointer,metaData) result(ref)
+      use pFIO
+      type(ArrayReference) :: ref
+      class(SwathGridFactory), intent(inout) :: this
+      real, pointer, intent(in) :: fpointer(:,:,:)
+      type(FileMetaData), intent(in), optional :: metaData
+      _UNUSED_DUMMY(this)
+      ref = ArrayReference(fpointer)
+   end function generate_file_reference3D
+
+  subroutine get_ncfile_dimension(filename, nlon, nlat, tdim)
+    use netcdf
+    implicit none
+    !
+    character(len=*), intent(in) :: filename
+    integer, intent(out) :: nlat, nlon, tdim
+    integer  :: ncid , iret, dimid
+    !
+    iret = nf90_open(trim(fileName), NF90_NOWRITE, ncid)
+    !
+    iret = nf90_inq_dimid(ncid, "time", dimid)
+    iret = nf90_inquire_dimension(ncid, dimid, len=tdim)
+    !
+    iret = nf90_inq_dimid(ncid, "lon", dimid)
+    iret = nf90_inquire_dimension(ncid, dimid, len=nlon)
+    !
+    iret = nf90_inq_dimid(ncid, "lat", dimid)
+    iret = nf90_inquire_dimension(ncid, dimid, len=nlat)
+    !
+    iret = nf90_close(ncid)
```

Comment:
These need checks on the value of `iret`:   `_VERIFY(iret)`

Reply:
I am getting compilation errors from:
    iret = nf90_open(trim(fileName), NF90_NOWRITE, ncid)
     _VERIFY(iret)
::
MAPL_SwathGridFactory.F90:1131:97:

 1131 |     _VERIFY(iret)
      |                                                                                                 1
Error: Symbol ‘rc’ at (1) has no IMPLICIT type

Reply:
Ah - the procedure that contains these lines does not itself have the usual GEOS return code (rc) optional argument.   And don't forget to add the `_RC` to the call into this procedure.

Reply:
Thanks, @tclune .  The error handling for netCDF is good, because when the netCDF error occurs, it can print out both the internal netCDF error message `nf90_strerror(status)` and print out the line number and the file name, using the macro _RC.